lib/flipper/test/shared_adapter_test.rb

```diff
@@ -0,0 +1,239 @@
+  end
+
+  def test_disables_all_enabled_things_when_boolean_gate_disabled
+    actor_22 = @actor_class.new('22')
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.boolean)
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.group(:admins))
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.actor(actor_22))
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.actors(25))
+    assert_equal true, @adapter.enable(@feature, @boolean_gate, @flipper.time(45))
+    assert_equal true, @adapter.disable(@feature, @boolean_gate, @flipper.boolean(false))
+    expected = {
+      :boolean => nil,
+      :groups => Set.new,
+      :actors => Set.new,
+      :percentage_of_actors => nil,
+      :percentage_of_time => nil,
+    }
+    assert_equal expected, @adapter.get(@feature)
+  end
+
+  def test_can_enable_disable_get_value_for_group_gate
+    assert_equal true, @adapter.enable(@feature, @group_gate, @flipper.group(:admins))
+    assert_equal true, @adapter.enable(@feature, @group_gate, @flipper.group(:early_access))
+
+    result = @adapter.get(@feature)
+    assert_equal Set['admins', 'early_access'], result[:groups]
+
+    assert_equal true, @adapter.disable(@feature, @group_gate, @flipper.group(:early_access))
+    result = @adapter.get(@feature)
+    assert_equal Set['admins'], result[:groups]
+
+    assert_equal true, @adapter.disable(@feature, @group_gate, @flipper.group(:admins))
+    result = @adapter.get(@feature)
+    assert_equal Set.new, result[:groups]
+  end
+
+  def test_can_enable_disable_and_get_value_for_an_actor_gate
+    actor_22 = @actor_class.new('22')
+    actor_asdf = @actor_class.new('asdf')
+
+    assert_equal true, @adapter.enable(@feature, @actor_gate, @flipper.actor(actor_22))
+    assert_equal true, @adapter.enable(@feature, @actor_gate, @flipper.actor(actor_asdf))
+
+    result = @adapter.get(@feature)
+    assert_equal Set['22', 'asdf'], result[:actors]
+
+    assert true, @adapter.disable(@feature, @actor_gate, @flipper.actor(actor_22))
+    result = @adapter.get(@feature)
+    assert_equal Set['asdf'], result[:actors]
+  end
+
+  def test_can_enable_disable_get_value_for_percentage_of_actors_gate
+    assert_equal true, @adapter.enable(@feature, @actors_gate, @flipper.actors(15))
+    result = @adapter.get(@feature)
+    assert_equal '15', result[:percentage_of_actors]
+
+    assert_equal true, @adapter.disable(@feature, @actors_gate, @flipper.actors(0))
+    result = @adapter.get(@feature)
+    assert_equal '0', result[:percentage_of_actors]
+  end
+
+  def test_can_enable_percentage_of_actors_gate_many_times_and_consistently_return_values
+    (1..100).each do |percentage|
+      assert_equal true, @adapter.enable(@feature, @actors_gate, @flipper.actors(percentage))
+      result = @adapter.get(@feature)
+      assert_equal percentage.to_s, result[:percentage_of_actors]
+    end
+  end
```

Comment:
Missing the disabled spec that matches this named "can disable percentage of actors gate many times and consistently return values".